pkg/apis/serving/v1alpha2/predictor.go

```diff
@@ -79,32 +78,11 @@ func (p *PredictorSpec) Validate(config *InferenceServicesConfig) error {
 	if err := validateReplicas(p.MinReplicas, p.MaxReplicas); err != nil {
 		return err
 	}
-	return nil
-}
-
-func validateStorageURI(storageURI string) error {
-	if storageURI == "" {
-		return nil
-	}
-
-	// local path (not some protocol?)
-	if !regexp.MustCompile("\\w+?://").MatchString(storageURI) {
-		return nil
-	}
-
-	// one of the prefixes we know?
-	for _, prefix := range SupportedStorageURIPrefixList {
-		if strings.HasPrefix(storageURI, prefix) {
-			return nil
-		}
-	}
-
-	azureURIMatcher := regexp.MustCompile(AzureBlobURIRegEx)
-	if parts := azureURIMatcher.FindStringSubmatch(storageURI); parts != nil {
-		return nil
+	if err := validateResourceRequirements(predictor.GetResourceRequirements()); err != nil {
```

Comment:
same here w/ loops.

Reply:
OK, will change all the places to a separate CL.